tests/test_georaster.py

```diff
@@ -1913,3 +1926,279 @@ def test_power(self, power: float | int) -> None:
+    ufuncs_str_1nin_2nout = [
+        ufunc for ufunc in ufuncs_str if (getattr(np, ufunc).nin == 1 and getattr(np, ufunc).nout == 2)
+    ]
+
+    # Universal functions with two input arguments and one output, corresponding to:
+    # ['add', 'arctan2', 'bitwise_and', 'bitwise_or', 'bitwise_xor', 'copysign', 'divide', 'equal', 'floor_divide',
+    #  'float_power', 'fmax', 'fmin', 'fmod', 'gcd', 'greater', 'greater_equal', 'heaviside', 'hypot', 'lcm', 'ldexp',
+    #  'left_shift', 'less', 'less_equal', 'logaddexp', 'logaddexp2', 'logical_and', 'logical_or', 'logical_xor',
+    #  'maximum', 'minimum', 'mod', 'multiply', 'nextafter', 'not_equal', 'power', 'remainder', 'right_shift',
+    #  'subtract', 'true_divide']
+    ufuncs_str_2nin_1nout = [
+        ufunc for ufunc in ufuncs_str if (getattr(np, ufunc).nin == 2 and getattr(np, ufunc).nout == 1)
+    ]
+
+    # Universal functions with two input arguments and two outputs (Note: none exist for three outputs or above)
+    # These correspond to: ['divmod']
+    ufuncs_str_2nin_2nout = [
+        ufunc for ufunc in ufuncs_str if (getattr(np, ufunc).nin == 2 and getattr(np, ufunc).nout == 2)
+    ]
+
+    # -- Second, we list array functions we intend to support in the array interface --
+
+    # To my knowledge, there is no list that includes all numpy functions (and we probably don't want to test them all)
+    # Let's include manually the important ones:
+    # - statistics: normal and for NaNs;
+    # - sorting and counting;
+    # Most other math functions are already universal functions
+
+    # The full list exists in Raster
+    handled_functions = gu.georaster.raster._HANDLED_FUNCTIONS
+
+    # Details below:
+    # NaN functions: [f for f in np.lib.nanfunctions.__all__]
+    # nanstatfuncs = ['nansum', 'nanmax', 'nanmin', 'nanargmax', 'nanargmin', 'nanmean', 'nanmedian', 'nanpercentile',
+    #             'nanvar', 'nanstd', 'nanprod', 'nancumsum', 'nancumprod', 'nanquantile']
+
+    # Statistics and sorting matching NaN functions: https://numpy.org/doc/stable/reference/routines.statistics.html
+    # and https://numpy.org/doc/stable/reference/routines.sort.html
+    # statfuncs = ['sum', 'max', 'min', 'argmax', 'argmin', 'mean', 'median', 'percentile', 'var', 'std', 'prod',
+    #              'cumsum', 'cumprod', 'quantile']
+
+    # Sorting and counting ounting with single array input:
+    # sortfuncs = ['sort', 'count_nonzero', 'unique]
+
+    # --  Third, we define the test data --
+
+    # We create two random array of varying dtype
+    width = height = 5
+    min_val = np.iinfo("int32").min
+    max_val = np.iinfo("int32").max
+    transform = rio.transform.from_bounds(0, 0, 1, 1, width, height)
+    np.random.seed(42)
+    arr1 = np.random.randint(min_val, max_val, (height, width), dtype="int32") + np.random.normal(size=(height, width))
+    arr2 = np.random.randint(min_val, max_val, (height, width), dtype="int32") + np.random.normal(size=(height, width))
```

Comment:
This makes me think. We must have written these few lines many times in the tests. If that's the case, maybe we could have a function to generate a random Raster of a given shape and dtype, to make sure we are consistent in all tests.

Reply:
Yes, it's not a CPU issue right now because we always define them outside the loops, so it's defined maybe ~3-4 times. But we could have it as a `pytest.fixture` to reuse it everywhere consistently, and ensuring it is not modified. Although it would make the code a bit harder to read...

Reply:
I leave as is and don't open an issue, and we have much more important things to do in the test suite as of now :sweat_smile:. 
A small future improvement for someone motivated.